include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -305,12 +305,26 @@ def TT_ExpandDimsOp : TT_Op<"expand_dims", [Pure,
 def TT_ReshapeOp : TT_Op<"reshape", [Pure,
                                      SameOperandsAndResultElementType]> {
     let summary = "reinterpret a tensor to a different shape. It may change elements order if the attribute is set.";
-    let arguments = (ins TT_Tensor:$src, BoolAttr:$allow_reorder);
+    let description = [{
+        reinterpret a tensor to a different shape.
+
+        If allow_reorder is set the compiler is free to change the order of
+        elements to generate more efficient code.
+
+        If efficient_layout is set, this is a hint that the destination layout should be kept for performance reason.
+        The compiler is still free to change it for better performance.
+    }];
+    let arguments = (ins TT_Tensor:$src, BoolAttr:$allow_reorder, OptionalAttr<UnitAttr>:$efficient_layout);
```

Comment:
It's confusing to me that allow_reorder is a boolean but efficient_layout is a unit.  I mean, I like the unit-ness, but it looks really inconsistent in the IR.

Reply:
Right, the reason I went for that was to make the IR printing slightly nicer by not printing the optional argument in case it is not set. While `allow_reorder` should always be specified I was seeing `efficient_layout` as an optional hint. 
What do you think? Maybe `allow_reorder` should also be an optional hint?

Reply:
I think I get your point -- the attribute doesn't mean anything until after the pass runs.  And I guess it's kind of misleading to write efficient_layout=False only to have the pass change it to True later.

I guess it might make some sense to make the attribute a tristate (i.e. true/false/not-present)?  This would make it explicit whether the pass has run and decided whether the layouts are efficient?  ("Explicit is better than implicit", https://peps.python.org/pep-0020/)

Or we can just leave it as-is.  Either way 🤷

Reply:
> I guess it might make some sense to make the attribute a tristate (i.e. true/false/not-present)

Allowing it be set but false seems more confusing to me. I don't have a strong opinion but in doubt I would leave it like that.